### PR TITLE
MWPW-137076 - Section metadata style - Grid-width-* doesn't respect blocks w/ .contained variant

### PR DESCRIPTION
**Pull request by ryanmparrish**
As an author in Milo, I want to have consistent display when using a section-metadata style: grid-width-8. etc. Or any of the grid-width-* styles (6, 8, 10)

Currently nested section blocks w/ the variant `.contained` don't respect the grid-width

Resolves: [MWPW-137076](https://jira.corp.adobe.com/browse/MWPW-137076)

**Test MILO URLs:**
Before:  (Notice the second set of icons (volusion, wix, woo.. don't respect grid-width-8)
https://main--milo--adobecom.hlx.page/drafts/rparrish/bacom/section-grid-width

After: 
https://rparrish-grid-width-fix--milo--adobecom.hlx.page/drafts/rparrish/bacom/section-grid-width

Before:
https://main--milo--adobecom.hlx.page/drafts/rparrish/bacom/digital-commerce

After:
https://rparrish-grid-width-fix--milo--adobecom.hlx.page/drafts/rparrish/bacom/digital-commerce

**LIVE BACOM URLs:**
Before
https://main--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce

After with milo branch 
https://main--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce?milolibs=rparrish-grid-width-fix

This PR will require some coordinating w/ a subsequent BACOM PR to remove the style patch this fixes. 

Before bacom branch 
https://rparrish-revert-grid-fix--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce

Revert-fix bacom branch 
https://rparrish-revert-grid-fix--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce?milolibs=rparrish-grid-width-fix

### CODE REVIEW

File: libs/blocks/section-metadata/section-metadata.css [hunk 1]
```diff
@@ -183,8 +183,14 @@
...
+.section.container[class*='-up'] {
+  width: auto;
+  margin: 0;
+}
+
 .section[class*='grid-width-'] > .content,
-main > .section[class*='-up'] > .content {
+main > .section[class*='-up'] > .content,
+.section[class*='grid-width-'] > .contained {
   max-width: initial;
```
Comment (JasonHowellSlavin) [inline libs/blocks/section-metadata/section-metadata.css:194]: unfortunately, this now overrides `max-width: var(--grid-container-width);` coming from [columns.css](https://github.com/adobecom/milo/blob/694389f75d3b1a5e01416f2f771a6246bb27102f/libs/blocks/columns/columns.css#L8) causing a bit of a margin error on mobile:
current mobile grid
fix mobile grid

File: libs/blocks/section-metadata/section-metadata.css [hunk 2]
```diff
@@ -183,8 +183,14 @@
   gap: var(--spacing-m);
 }
 
+.section.container[class*='-up'] {
+  width: auto;
+  margin: 0;
+}
```
[the remark this answers is not in the document]
Comment (ryanmparrish) [reply libs/blocks/section-metadata/section-metadata.css]: Sorry, I left this sit so long. 
I removed this selector. It wasn't intended to be added Sorry for the confusion.

**Comment by JasonHowellSlavin** [comment]
> Figured it's worth noting since there have been commits since @JasonHowellSlavin 's comment -- I'm still seeing weird alignment on that page he screenshotted, just in relation to the icon group centering on the page margin, not the individual icons centering around each other.
> 
> If this is due to something unrelated to this PR though, just let me know and I'll drop an approval here 👍
> 

I might check with our PMs how many pages have those icon blocks. They were the initial reason we added overrides I believe. Maybe if the authoring is not too much, we can see if they can change them. Might solve a few of our issues actually.

**Comment by ryanmparrish** [comment]
@elan-tbx  - Thanks for bumping this. For a li more context. This is a two part PR so you can really only see the final results on both the bacom feature with the milo branch url param. [https://rparrish-revert-grid-fix--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce?milolibs=rparrish-grid-width-fix]

@JasonHowellSlavin, @meganthecoder  - Can we sync up on this one and A) See if it's still needed B) if so, coordinate on the tandem PRs in each repo?

**Comment by JasonHowellSlavin** [comment]
> @elan-tbx - Thanks for bumping this. For a li more context. This is a two part PR so you can really only see the final results on both the bacom feature with the milo branch url param. [https://rparrish-revert-grid-fix--bacom--adobecom.hlx.page/products/experience-manager/digital-commerce?milolibs=rparrish-grid-width-fix]
> 
> @JasonHowellSlavin, @meganthecoder - Can we sync up on this one and A) See if it's still needed B) if so, coordinate on the tandem PRs in each repo?

Definitely. I am reaching out to our authors to also see if we can't fix what the initial cause of the overrides were. I'd like to remove them, and if this gets us 99% of the way there but we need to update some pages, that might be what is has to be.

**Pull request closed without merging**